docs/source/evaluation_scripts.md

```diff
@@ -52,6 +52,6 @@ output['result'] = [
 
 ```
 
-`output` should contain a key named `result`, which is a list containing entries per challenge phase split. Each challenge phase split object contains various keys, which are then displayed as columns in leaderboard.
+`output` should contain a key named `result`, which is a list containing entries per challenge phase split. Each of these entries will have a key with the corresponding challenge phase codename. For this example, `challenge_phase_split_1` and `challenge_phase_split_2` are codenames for challenge phases. Each challenge phase split object contains various keys (`score` in this example), which are then displayed as columns in the leaderboard. 
```

Comment:
@live-wire it looks like there is some confusion. The `challenge_phase_split_1` and `challenge_phase_split_2` are the dataset split ids instead of challenge phase split ids. Moreover, there is no such challenge phase split id in the configuration of a challenge. Can you please make these edits wherever you find the inconsistency? 

I am glad that you brought up this point. Thanks!

Reply:
Hi @deshraj I was also confused when I read names challenge phase split. 
But, to be sure, I re-checked the sample evaluation script for the VQA challenge. 
The sample evaluation script was zipped, so I created a copy of the extracted file [here](https://gist.github.com/live-wire/c1098e788a94ce32ebdd97af716dd7b8). 

Our concern is what the keys `challenge_phase_split_1` and `challenge_phase_split_2` are in the following block:
```
output['result'] = [
    {
        'challege_phase_split_1': {
            'score': score,
        }
    },
    {
        'challege_phase_split_2': {
            'score': score,
        }
    }
]

```
Please correct me if I'm wrong. 
If I start going back from where these keys get populated, I would start at [this](https://gist.github.com/live-wire/c1098e788a94ce32ebdd97af716dd7b8#file-main-py-L229) line, and then follow this order going back: 
[keys populated in results array](https://gist.github.com/live-wire/c1098e788a94ce32ebdd97af716dd7b8#file-main-py-L205),  [keys obtained from this](https://gist.github.com/live-wire/c1098e788a94ce32ebdd97af716dd7b8#file-main-py-L160) and finally [this](https://gist.github.com/live-wire/c1098e788a94ce32ebdd97af716dd7b8#file-main-py-L40). 
The keys in that dictionary are `train-dev2015`, `train-challenge2015` etc. which seem to be challenge phase codenames judging by [this](https://github.com/Cloud-CV/EvalAI-Examples/blob/master/vqa_challenge/vqa_challenge/vqa_challenge.yaml#L18) yaml configuration file for the same challenge.
Do I make sense ? I'm a little confused here. 👶

Reply:
Sorry, @live-wire I misinterpreted your changes and did the wrong comment. Please see the updated comment here:

Short answer: `challenge_phase_split_1` and `challenge_phase_split_2` are dataset split codenames associated with the phase to which submissions is done. 

Long answer: To clarify the doubt, please see the below `evaluation()` method for example:

```python
def evaluate(test_annotation_file, user_annotation_file, phase_codename, **kwargs):
    score = (randint(0, 99))

    output = {}
    if phase_codename == "dev":
        output['result'] = [
            {
                'train_split': {
                    'score': score,
                }
            },
        ]
    elif phase_codename == "test":
        output['result'] = [
            {
                'train_split': {
                    'score': score,
                }
            },
            {
                'test_split': {
                    'score': score,
                }
            }
        ]
    return output
```

In the above example, `dev` and `test` are two different challenge phases. There are two dataset splits `train_split` and `test_split`. Note that the `dev` phase just accepts submissions for one split i.e `train_split` whereas `test` phase accepts submissions to both `train_split` and `test_split`. Let me know if this clears your doubt. 

For even better understanding, see the challenge config zip attached in #1590. 

You might want to change the documentation now to this example to make it better understandable. 

Thanks again for raising this concern.

Reply:
This clears it up really well. 😄  I'll update the documentation and the zip submission I made [here](https://github.com/Cloud-CV/EvalAI/pull/1591) with these changes 👍